Source/quests.cpp

```diff
@@ -283,31 +285,33 @@ void InitQuests()
 	WaterDone = 0;
 	int initiatedQuests = 0;
 
-	for (int i = 0; i < MAXQUESTS; i++) {
-		if (gbIsMultiplayer && QuestData[i].isSinglePlayerOnly)
+	int q = 0;
+	for (auto &quest : Quests) {
+		if (gbIsMultiplayer && QuestData[q].isSinglePlayerOnly)
 			continue;
-		Quests[i]._qtype = QuestData[i]._qdtype;
+		quest._qtype = QuestData[q]._qdtype;
 		if (gbIsMultiplayer) {
-			Quests[i]._qlevel = QuestData[i]._qdmultlvl;
+			quest._qlevel = QuestData[q]._qdmultlvl;
 			if (!delta_quest_inited(initiatedQuests)) {
-				Quests[i]._qactive = QUEST_INIT;
-				Quests[i]._qvar1 = 0;
-				Quests[i]._qlog = false;
+				quest._qactive = QUEST_INIT;
+				quest._qvar1 = 0;
+				quest._qlog = false;
 			}
 			initiatedQuests++;
 		} else {
-			Quests[i]._qactive = QUEST_INIT;
-			Quests[i]._qlevel = QuestData[i]._qdlvl;
-			Quests[i]._qvar1 = 0;
-			Quests[i]._qlog = false;
+			quest._qactive = QUEST_INIT;
+			quest._qlevel = QuestData[q]._qdlvl;
+			quest._qvar1 = 0;
+			quest._qlog = false;
 		}
 
-		Quests[i]._qslvl = QuestData[i]._qslvl;
-		Quests[i].position = { 0, 0 };
-		Quests[i]._qidx = i;
-		Quests[i]._qlvltype = QuestData[i]._qlvlt;
-		Quests[i]._qvar2 = 0;
-		Quests[i]._qmsg = QuestData[i]._qdmsg;
+		quest._qslvl = QuestData[q]._qslvl;
+		quest.position = { 0, 0 };
+		quest._qidx = q;
+		quest._qlvltype = QuestData[q]._qlvlt;
+		quest._qvar2 = 0;
+		quest._qmsg = QuestData[q]._qdmsg;
+		q++;

```

Comment:
This loop has a continue condition so this would not be executed and would miss-align in multiplayer.